src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -100,5 +100,9 @@ static void CloseConnection(AmqpConnection connection)
 
             return connection;
         }
+
+        /// <summary>Namespace name used for <see cref="ServiceBusConnection"/>.</summary>
+        /// <value>Namespace name.</value>
+        internal string NamespaceName => this.Endpoint.Host.Split(new[] {'.'}, StringSplitOptions.RemoveEmptyEntries)[0];
```

Comment:
Yes. That's right.
We have public domains as well as separate national clouds. The domain name `servicebus.windows.net` should not be assumed.